### PR TITLE
[FLINK-23558][streaming] Ignoring RejectedExecutionException during s…

### PR DESCRIPTION
**Pull request by akalash**
…ubmitting the throughput calculation

## What is the purpose of the change

*This PR fixes result of the race condition between submitting the throughput calculation and finishing the task. *

## Brief change log
  - *Ignoring RejectedExecutionException during submitting the throughput calculation*

## Verifying this change

This change added tests and can be verified as follows:

  - *Test for StreamTask*
## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (yes / **no**)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (yes / **no**)
  - The serializers: (yes / **no** / don't know)
  - The runtime per-record code paths (performance sensitive): (yes / **no** / don't know)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: (yes / **no** / don't know)
  - The S3 file system connector: (yes / **no** / don't know)

## Documentation

  - Does this pull request introduce a new feature? (yes / **no**)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)

### CODE REVIEW

**Comment by curcur** [comment]
I have a bit concern of this:

this change swallows all "RejectedExecutionException" no matter it is because of task finish or not.

I think the right fix is not to submit mailbox execution after the task finishes?

**Comment by akalash** [comment]
@flinkbot run azure

**Comment by curcur** [comment]
@akalash 

I've looked through the code again, I think we should probably also keep 
```      
        timersFinishedFuture.get();
        systemTimersFinishedFuture.get();
```

in the task thread for the safety purposes. 

I mean we move this two lines of code into the actionExecutor and keep the same copy in its origin place.

I think for normal task, actionExecutor is the same task thread, but there is also some cases (like sources) that they maybe  different

**Review by curcur — APPROVED** [review]

**Comment by curcur** [comment]
Thanks @akalash for fixing this. Approved with a bit of comments.

File: flink-streaming-java/src/test/java/org/apache/flink/streaming/runtime/tasks/StreamTaskTest.java [hunk 1]
```diff
@@ -1635,6 +1636,56 @@ public void testRestorePerformedFromInvoke() throws Exception {
...
+                            stoppintMailboxLatch.countDown();
+                            task.streamTask.afterInvoke();
+                        },
+                        "test")
+                .get();
+
+        // then: the exception handle wasn't invoked because the such situation is expected.
+        assertFalse(submitThroughputFail.get());
+
+        inputProcessor.availabilityProvider.getUnavailableToResetAvailable().complete(null);
```
[the remark this answers is not in the document]
Comment (curcur) [reply flink-streaming-java/src/test/java/org/apache/flink/streaming/runtime/tasks/StreamTaskTest.java:1686]: I mean the last line.

**Comment by curcur** [comment]
Thanks for fixing this @akalash , merged.

**Pull request merged**